axum-macros/src/lib.rs

```diff
@@ -513,12 +513,29 @@ pub fn derive_from_request(item: TokenStream) -> TokenStream {
 /// async fn handler(request: Request<BoxBody>) {}
 /// ```
 ///
+/// # Changing state type
+///
+/// By default `#[debug_handler]` assumes your state type is `()`. Thus if you're using
+/// [`axum::extract::State`] you must change the state type to match:
+///
+/// ```
+/// use axum::extract::State;
+/// # use axum_macros::debug_handler;
+///
+/// #[debug_handler(state = AppState)]
+/// async fn handler(State(staste): State<AppState>) {}
```

Comment:
Why not parse the argument / only have this state attribute as a fallback for indirect state type requirements (through other extractors / nested uses of `State`)?

Reply:
Sure!